src/test/java/com/microsoft/graph/functional/OneNoteTests.java

```diff
@@ -429,6 +437,65 @@ public void testMultipartPost(){
             fail("Unable to write to output stream");
         }
     }
+    
+    /**
+     * Test posting multipart content to a page
+     */
+    @Test
+    public void testMultipartPostWithHeadersMap() throws Exception{
+    	Multipart multipart = new Multipart();
+
+    	String htmlContent = "<!DOCTYPE html>\r\n" +
+    			"<html lang=\"en-US\">\r\n" +
+    			"<head>\r\n" +
+    			"<title>Test Multipart Page</title>\r\n" +
+    			"<meta name=\"created\" content=\"2001-01-01T01:01+0100\">\r\n" +
+    			"</head>\r\n" +
+    			"<body>\r\n" +
+    			"<p>\r\n" +
+    			"<img src=\"name:image\" />\r\n" +
+    			"</p>\r\n" +
+    			"<p>\r\n" +
+    			"<object data=\"name:attachment\" data-attachment=\"document.pdf\" /></p>\r\n" +
+    			"\r\n" +
+    			"</body>\r\n" +
+    			"</html>";
+    	File imgFile = new File("src/test/resources/hamilton.jpg");
+    	File pdfFile = new File("src/test/resources/document.pdf");
+
+    	Map<String, String> htmlHeaderMap = new HashMap<>();
+    	Map<String, String> contentDispMap = new HashMap<>();
+    	contentDispMap.put("name","Presentation" );
+    	htmlHeaderMap.put("Content-Disposition", Multipart.createContentHeaderValue("form-data", contentDispMap));
+    	htmlHeaderMap.put("Content-Type", Multipart.createContentHeaderValue("text/html", null));
+    	multipart.addPart(htmlHeaderMap, htmlContent.getBytes(HTML_ENCODING));
+
+    	InputStream fileStream = new FileInputStream(imgFile);
+    	multipart.addFormData("hamilton", "image/jpg", getByteArray(fileStream));
+    	multipart.addFilePart("metadata", "application/pdf", pdfFile);
+
+    	// Add multipart request header
+    	List<Option> options = new ArrayList<Option>();
+    	options.add(multipart.header());
+
+    	// Post the multipart content
+    	IOnenotePageCollectionRequestBuilder pageReq = orb
+    			.sections(testSection.id)
+    			.pages();
+    	String expectedRequestUrl = "https://graph.microsoft.com/v1.0/me/onenote/sections/"+testSection.id+"/pages";
+    	assertEquals(expectedRequestUrl, pageReq.getRequestUrl());
+    	IOnenotePageCollectionRequest request = pageReq.buildRequest(options);
+    	assertNotNull(request);
+    	OnenotePageCollectionRequest pageCollectionReq = (OnenotePageCollectionRequest)request;
+    	List<HeaderOption> headeroption = pageCollectionReq.getHeaders();
+    	assertEquals("Content-Type", headeroption.get(0).getName());
+    	String expectedHeaderValue = "multipart/form-data; boundary=\""+multipart.getBoundary()+"\"";
+    	assertEquals(expectedHeaderValue, headeroption.get(0).getValue().toString());
+    	assertNotNull(multipart.content());
+    	OnenotePage page = pageReq.buildRequest(options)
```

Comment:
you have already taken out request object in line 487, why making pageReq.buildRequest(options) call again?

Reply:
changed it to use request.post(...)